src/applications/personalization/profile-2/components/connected-apps/ConnectedApps.jsx

```diff
@@ -43,7 +43,12 @@ export class ConnectedApps extends Component {
     const activeApps = apps ? apps.filter(app => !app.deleted) : [];
 
     const allAppsDeleted = deletedApps?.length === apps?.length;
-    const showHasNoConnectedApps = !apps || (allAppsDeleted && !loading);
+    // We treat this 404 'Record not found' error as an empty apps array
+    const firstError = errors?.[0];
+    const recordNotFound =
+      firstError?.title === 'Record not found' && firstError?.status === '404';
```

Comment:
I really don't know the best way to approach this, but my instinct was to just check the status and ignore the title.

Reply:
I decided on checking for both because I thought we might run into another 404 with a different 'Not found' title and we only want to treat this specific 404 as a non-error.